packages/better-php-doc-parser/src/AnnotationReader/NodeAnnotationReader.php

```diff
@@ -87,13 +96,14 @@ public function readClassAnnotation(Class_ $class, string $annotationClassName):
     public function readPropertyAnnotation(Property $property, string $annotationClassName): ?object
     {
         $propertyReflection = $this->createPropertyReflectionFromPropertyNode($property);
-        if (! $propertyReflection instanceof ReflectionProperty) {
+        if (! $propertyReflection instanceof PhpPropertyReflection) {
             return null;
         }
 
         try {
             // covers cases like https://github.com/rectorphp/rector/issues/3046
 
+            // @todo this will require original reflection
             /** @var object[] $propertyAnnotations */
             $propertyAnnotations = $this->reader->getPropertyAnnotations($propertyReflection);
             return $this->matchNextAnnotation($propertyAnnotations, $annotationClassName, $property);
```

Comment:
@ondrejmirtes This one is tricky :/ we're using Doctrine annotation reader that work with native `PropertyReflection`. I could not find the native reflection in `PhpPropertyReflection`. Any idea how to work around this?

Reply:
Does this work for you? https://github.com/phpstan/phpstan-src/commit/5fad625b7770b9c5beebb19ccc1a493839308fb4

Reply:
It does! The annotation is resolved correctly :+1: 

(demo)
https://github.com/rectorphp/rector/pull/5665/commits/d896421b74d2b4b67218611c9e79667ba3edd988#diff-2dc622f75265883d5433d11b29facf32d8f3a0bc244b3f4822f6a3881ad2fa7aR212

Reply:
Is there is a similar method for class and method reflection? The reader needs them:

```php
$this->reader->getClassAnnotations()
$this->reader->getMethodAnnotations()
```

Reply:
`ClassReflection::getNativeReflection()`. You can actually get native `ReflectionMethod` on that object as well :)

Reply:
Via `getNativeMethod()`. Cool :+1: